Add plugin: Mouse Navigation

HoBeom opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [X]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [X] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [X] `main.js`
  - [X] `manifest.json`
  - [x] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] My README.md describes the plugin's purpose and provides clear usage instructions.
- [X] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [X] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [X] I have added a license in the LICENSE file.
- [X] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["repo": "hobeom/obsidian-mouse-navigation",](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/manifest.json#L8)
Not a valid part of the manifest, remove it.

[mySetting: string;](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L12), [mySetting: 'default',](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L26)
Remove this sample code

[actionText = 'Next Page';](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L300), [actionText = 'Previous Page';](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L303), [actionText = 'Scroll to Top';](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L306), [actionText = 'Scroll to Bottom';](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L309), ['Right Click'](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L409), [.setName('Enable Drawing')](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L421), [.setName('Stroke Color')](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L434), ['accent': 'Accent Color',](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L439), [setName('Custom Stroke Color')](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L460), [.setName('Line Width')](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L475)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[const currentView = this.app.workspace.getActiveViewOfType(](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L335-L337)
Change this to:
```ts
const currentView = this.app.workspace.getActiveViewOfType(
	View,
);
```

[if (currentView == null || currentView?.file?.extension == 'md') {](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L343)
And this to
```ts
if (currentView == null || !(currentView instanceof EditableFileView) || currentView?.file?.extension == 'md') {
```

[const preview = markdownView.previewMode;](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L352)
There are a bunch of places where you are not checking that `markdownView` is not null.

[(this.app as unknown)](https://github.com/HoBeom/obsidian-mouse-navigation/blob/e9f7b02a20365d3700d1a77e27093b20611255a4/main.ts#L373)
Why this cast?

HoBeom commented:
I have updated the code according to the feedback. The scroll function has been improved, and I have added null checks for `markdownView` to prevent forced termination in new tabs.

I would also like to seek assistance regarding the context menu issue mentioned in [this issue](
https://github.com/HoBeom/obsidian-mouse-navigation/issues/1 ). With the plugin activated, the right-click menus on the previewmode don't appear, which prevents access to many options (also auto word selection). Any guidance would be greatly appreciated.
